File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/mathrax/domain/MathraxObject.java
package com.zwstudio.logicpuzzlesandroid.puzzles.mathrax.domain;

/**
 * Created by TCC-2-9002 on 2016/10/20.
 */

public abstract class MathraxObject {
    public abstract String objAsString();
    public static MathraxObject objFromString(String str) {
        switch (str) {
            case "marker":
                return new MathraxMarkerObject();
            case "tree":
                return new MathraxTreeObject();
            default:
                return new MathraxEmptyObject();
        }
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/mathrax/MathraxObject.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.mathrax

import com.zwstudio.logicpuzzlesandroid.common.domain.Position

class MathraxHint(var op: Char = ' ', var result: Int = 0) {
    override fun toString() = (if (result == 0) "" else result.toString()) + op.toString()
}

class MathraxGameMove(val p: Position, var obj: Int = 0)
